src/main/java/seedu/address/model/calendar/UniqueMeetingList.java

```diff
@@ -1,4 +1,141 @@
 package seedu.address.model.calendar;
 
-public class UniqueMeetingList {
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.calendar.exceptions.DuplicateMeetingException;
+import seedu.address.model.calendar.exceptions.MeetingNotFoundException;
+import seedu.address.model.property.Property;
+
+
+/**
+ * A list of meetings that enforces uniqueness between its elements and does not allow nulls.
+ * A meeting is considered unique by comparing using {@code Meeting#isSameMeeting(Meeting)}.
+ * As such, adding and updating of meetings uses Meeting#isSameMeeting(Meeting) for equality so as
+ * to ensure that the meeting being added or updated is unique in terms of identity in the
+ * UniqueMeetingList. However, the removal of a property uses Property#equals(Object) so as to ensure
+ * that the property with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Property#isSameProperty(Property)
+ */
+public class UniqueMeetingList implements Iterable<CalendarMeeting> {
+
+    private final ObservableList<CalendarMeeting> internalList = FXCollections.observableArrayList();
+    private final ObservableList<CalendarMeeting> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent Meeting as the given argument.
+     */
+    public boolean contains(CalendarMeeting toCheck) {
```

Comment:
I think it would be better to change the naming of the CalendarMeeitng class to Meeting as it makes the code unnecessarily long and there are not other places where this is used.